og.module

```diff
@@ -505,15 +505,30 @@ function og_node_access($node, $op, $account) {
     // there is no group context. However, we can check if there are any groups
     // the user will be able to select, and if not, we don't allow access.
     // @see OgSelectionHandler::getReferencableEntities()
+    $accesses = array();
     $required = FALSE;
     foreach (og_get_group_audience_fields('node', $type) as $field_name => $label) {
       $field = field_info_field($field_name);
       $instance = field_info_instance('node', $field_name, $type);
       // Set the "field mode" to default, before passing it to the
       // selection handler.
       $instance['field_mode'] = 'default';
-      if (entityreference_get_selection_handler($field, $instance)->countReferencableEntities()) {
-        return NODE_ACCESS_ALLOW;
+      if (is_object($node) && isset($node->{$field_name})) {
+        if (!empty($node->{$field_name}) && !empty($node->{$field_name}[LANGUAGE_NONE])) {
+          foreach ($node->{$field_name}[LANGUAGE_NONE] as $target_id) {
+            if (og_user_access('node', $target_id['target_id'], 'create '.$type.' content')) {
+              $accesses[] = NODE_ACCESS_ALLOW;
+            }
+            else {
+              $accesses[] = NODE_ACCESS_DENY;
+            }
+          }
+        }
+      }
+      else {
```

Comment:
elseif ?

Reply:
I think this should actually use else...